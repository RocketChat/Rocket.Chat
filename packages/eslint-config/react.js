/** @type {import('eslint').ESLint.ConfigData} */
const config = {
	plugins: ['react', 'react-hooks', 'jsx-a11y'],
	extends: ['plugin:jsx-a11y/recommended'],
	rules: {
		'react-hooks/exhaustive-deps': 'error',
		'react-hooks/rules-of-hooks': 'error',
		'react/display-name': 'error',
		'react/jsx-curly-brace-presence': 'error',
		'react/jsx-fragments': ['error', 'syntax'],
		'react/jsx-key': ['error', { checkFragmentShorthand: true, checkKeyMustBeforeSpread: true, warnOnDuplicates: true }],
		'react/jsx-no-undef': 'error',
		'react/jsx-uses-react': 'error',
		'react/jsx-uses-vars': 'error',
		'react/no-multi-comp': 'error',
		'jsx-a11y/no-autofocus': [2, { ignoreNonDOM: true }],
	},
	settings: {
		react: {
			version: 'detect',
		},
	},
	overrides: [
		{
			files: ['**/*.stories.js', '**/*.stories.jsx', '**/*.stories.ts', '**/*.stories.tsx', '**/*.spec.tsx'],
			rules: {
				'react/display-name': 'off',
				'react/no-multi-comp': 'off',
			},
		},
	],
};

module.exports = config;
