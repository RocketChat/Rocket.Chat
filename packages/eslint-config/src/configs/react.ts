import jsxA11y from 'eslint-plugin-jsx-a11y';
import reactPlugin from 'eslint-plugin-react';
import reactHooks from 'eslint-plugin-react-hooks';
import reactRefresh from 'eslint-plugin-react-refresh';
import testingLibrary from 'eslint-plugin-testing-library';

import globals from 'globals';
import type { Linter } from 'eslint';

const configs = {
	'testing-library': {
		files: ['**/*.stories.js', '**/*.stories.jsx', '**/*.stories.ts', '**/*.stories.tsx', '**/*.spec.tsx'],
		plugins: {
			'testing-library': testingLibrary,
		},
		rules: {
			'testing-library/await-async-events': ['warn', { eventModule: 'userEvent' }],
			'testing-library/await-async-queries': 'warn',
			'testing-library/await-async-utils': 'warn',
			'testing-library/no-await-sync-events': ['warn', { eventModules: ['fire-event'] }],
			'testing-library/no-await-sync-queries': 'warn',
			'testing-library/no-container': 'warn',
			'testing-library/no-debugging-utils': 'warn',
			'testing-library/no-dom-import': ['warn', 'react'],
			'testing-library/no-global-regexp-flag-in-query': 'warn',
			'testing-library/no-manual-cleanup': 'warn',
			'testing-library/no-node-access': 'warn',
			'testing-library/no-promise-in-fire-event': 'warn',
			'testing-library/no-render-in-lifecycle': 'warn',
			'testing-library/no-unnecessary-act': 'warn',
			'testing-library/no-wait-for-multiple-assertions': 'warn',
			'testing-library/no-wait-for-side-effects': 'warn',
			'testing-library/no-wait-for-snapshot': 'warn',
			'testing-library/prefer-find-by': 'warn',
			'testing-library/prefer-presence-queries': 'warn',
			'testing-library/prefer-query-by-disappearance': 'warn',
			'testing-library/prefer-screen-queries': 'warn',
			'testing-library/render-result-naming-convention': 'warn',
		},
	},
	'jsx-a11y': {
		plugins: {
			// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
			'jsx-a11y': jsxA11y,
		},
		rules: {
			'jsx-a11y/click-events-have-key-events': 'warn',
			'jsx-a11y/no-static-element-interactions': 'warn',
			'jsx-a11y/no-autofocus': 'warn',
			'jsx-a11y/media-has-caption': 'warn',
			'jsx-a11y/iframe-has-title': 'warn',
			'jsx-a11y/alt-text': 'warn',
		},
		settings: {
			react: {
				version: 'detect',
			},
		},
	},
} satisfies Record<string, Linter.Config>;

type ReactRules = {
	[Key in keyof (typeof reactPlugin)['rules'] as `react/${Key}`]?: Linter.RuleEntry<unknown[]>;
};

export default function react(config: Linter.Config<ReactRules> = {}): Linter.Config[] {
	return [
		reactPlugin.configs.flat['recommended']!,
		reactPlugin.configs.flat['jsx-runtime']!,
		reactHooks.configs['recommended-latest'],
		reactRefresh.configs.recommended,
		configs['jsx-a11y'],
		configs['testing-library'],
		{
			languageOptions: {
				globals: globals.browser,
			},
		},
		config,
	];
}
