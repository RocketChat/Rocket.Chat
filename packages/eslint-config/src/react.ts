import tseslint, { type ConfigWithExtends } from 'typescript-eslint';
import jsxA11y from 'eslint-plugin-jsx-a11y';
import reactPlugin from 'eslint-plugin-react';
import reactHooks from 'eslint-plugin-react-hooks';
import reactRefresh from 'eslint-plugin-react-refresh';
import testingLibrary from 'eslint-plugin-testing-library';
import storybook from 'eslint-plugin-storybook';
import globals from 'globals';
import type { FlatConfig } from '@typescript-eslint/utils/ts-eslint';

const configs = {
	'testing-library': {
		files: ['**/*.stories.js', '**/*.stories.jsx', '**/*.stories.ts', '**/*.stories.tsx', '**/*.spec.tsx'],
		plugins: {
			'testing-library': testingLibrary,
		},
		rules: {
			'testing-library/await-async-events': ['warn', { eventModule: 'userEvent' }],
			'testing-library/await-async-queries': 'warn',
			'testing-library/await-async-utils': 'warn',
			'testing-library/no-await-sync-events': ['warn', { eventModules: ['fire-event'] }],
			'testing-library/no-await-sync-queries': 'warn',
			'testing-library/no-container': 'warn',
			'testing-library/no-debugging-utils': 'warn',
			'testing-library/no-dom-import': ['warn', 'react'],
			'testing-library/no-global-regexp-flag-in-query': 'warn',
			'testing-library/no-manual-cleanup': 'warn',
			'testing-library/no-node-access': 'warn',
			'testing-library/no-promise-in-fire-event': 'warn',
			'testing-library/no-render-in-lifecycle': 'warn',
			'testing-library/no-unnecessary-act': 'warn',
			'testing-library/no-wait-for-multiple-assertions': 'warn',
			'testing-library/no-wait-for-side-effects': 'warn',
			'testing-library/no-wait-for-snapshot': 'warn',
			'testing-library/prefer-find-by': 'warn',
			'testing-library/prefer-presence-queries': 'warn',
			'testing-library/prefer-query-by-disappearance': 'warn',
			'testing-library/prefer-screen-queries': 'warn',
			'testing-library/render-result-naming-convention': 'warn',
		},
	},
	'jsx-a11y': {
		plugins: {
			'jsx-a11y': jsxA11y,
		},
		rules: {
			'jsx-a11y/click-events-have-key-events': 'warn',
			'jsx-a11y/no-static-element-interactions': 'warn',
			'jsx-a11y/no-autofocus': 'warn',
			'jsx-a11y/media-has-caption': 'warn',
			'jsx-a11y/iframe-has-title': 'warn',
			'jsx-a11y/alt-text': 'warn',
		},
		settings: {
			react: {
				version: 'detect',
			},
		},
	},
} satisfies Record<string, ConfigWithExtends>;

function react({ languageOptions, ...config }: ConfigWithExtends = {}): FlatConfig.ConfigArray {
	return tseslint.config(
		{
			plugins: {
				react: reactPlugin,
			},
			languageOptions: {
				parserOptions: {
					ecmaFeatures: {
						jsx: true,
					},
					jsxPragma: null,
				},
			},
			rules: {
				...reactPlugin.configs.flat.recommended.rules,
				...reactPlugin.configs.flat['jsx-runtime'].rules,
				'react/no-unescaped-entities': 'warn',
				'react/no-children-prop': 'warn',
				'react/prop-types': 'warn',
				'react/display-name': 'warn',
				'react/jsx-curly-brace-presence': 'warn',
				'react/jsx-fragments': ['warn', 'syntax'],
				'react/jsx-key': ['warn', { checkFragmentShorthand: true, checkKeyMustBeforeSpread: true, warnOnDuplicates: true }],
				'react/jsx-no-undef': 'warn',
				'react/jsx-uses-vars': 'warn',
				'react/jsx-no-target-blank': 'warn',
				'react/no-multi-comp': 'warn',
				'react/no-direct-mutation-state': 'warn',
				'react/no-unknown-property': 'warn',
			},
		},
		reactHooks.configs['recommended-latest'],
		{
			rules: {
				'react-hooks/exhaustive-deps': 'warn',
				'react-hooks/rules-of-hooks': 'warn',
			},
		},
		reactRefresh.configs.recommended,
		{
			rules: {
				'react-refresh/only-export-components': 'warn',
			},
		},
		storybook.configs['flat/recommended'],
		{
			rules: {
				'storybook/no-renderer-packages': 'warn',
			},
		},
		configs['jsx-a11y'],
		configs['testing-library'],
		{
			languageOptions: {
				globals: globals.browser,
				...languageOptions,
			},
		},
		config,
	);
}

export default react;
