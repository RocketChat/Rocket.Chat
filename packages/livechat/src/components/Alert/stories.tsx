import { action } from '@storybook/addon-actions';
import type { Meta, StoryFn } from '@storybook/preact';
import type { ComponentProps } from 'preact';

import Alert from '.';
import { loremIpsum, screenDecorator } from '../../../.storybook/helpers';

export default {
	title: 'Components/Alert',
	component: Alert,
	args: {
		success: false,
		warning: false,
		error: false,
		color: '',
		timeout: 5000,
		onDismiss: action('dismiss'),
		children: loremIpsum({ count: 3, units: 'words' }),
	},
	parameters: {
		layout: 'centered',
	},
	decorators: [screenDecorator],
} satisfies Meta<ComponentProps<typeof Alert>>;

const Template: StoryFn<ComponentProps<typeof Alert>> = (args) => <Alert {...args} />;

export const Default = Template.bind({});
Default.storyName = 'default';
Default.args = {};

export const Success = Template.bind({});
Success.storyName = 'success';
Success.args = {
	success: true,
};

export const Warning = Template.bind({});
Warning.storyName = 'warning';
Warning.args = {
	warning: true,
};

export const Error = Template.bind({});
Error.storyName = 'error';
Error.args = {
	error: true,
};

export const CustomColor = Template.bind({});
CustomColor.storyName = 'custom color';
CustomColor.args = {
	color: '#175CC4',
};

export const WithLongTextContent = Template.bind({});
WithLongTextContent.storyName = 'with long text content';
WithLongTextContent.args = {
	children: loremIpsum({ count: 30, units: 'words' }),
};

export const WithoutTimeout = Template.bind({});
WithoutTimeout.storyName = 'without timeout';
WithoutTimeout.args = {
	timeout: 0,
};
