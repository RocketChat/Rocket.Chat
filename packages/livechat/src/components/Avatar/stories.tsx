import type { Meta, StoryFn } from '@storybook/preact';
import type { ComponentProps } from 'preact';

import { Avatar } from '.';
import { gazzoAvatar } from '../../../.storybook/helpers';

export default {
	title: 'Components/Avatar',
	component: Avatar,
	args: {
		src: gazzoAvatar,
		description: 'user description',
		status: undefined,
		large: false,
	},
	argTypes: {
		status: {
			control: {
				type: 'select',
				options: [undefined, 'offline', 'away', 'busy', 'online'],
			},
		},
	},
	parameters: {
		layout: 'centered',
	},
} satisfies Meta<ComponentProps<typeof Avatar>>;

const Template: StoryFn<ComponentProps<typeof Avatar>> = (args) => <Avatar {...args} />;

export const Default = Template.bind({});
Default.storyName = 'default';

export const Large = Template.bind({});
Large.storyName = 'large';
Large.args = {
	large: true,
};

export const Small = Template.bind({});
Small.storyName = 'small';
Small.args = {
	small: true,
};

export const AsPlaceholder: StoryFn<ComponentProps<typeof Avatar>> = (args) => (
	<div style={{ display: 'flex', alignItems: 'center', justifyContent: 'space-around' }}>
		<Avatar {...args} style={{ margin: '0.5rem' }} large />
		<Avatar {...args} style={{ margin: '0.5rem' }} />
		<Avatar {...args} style={{ margin: '0.5rem' }} small />
	</div>
);
AsPlaceholder.storyName = 'as placeholder';
AsPlaceholder.args = {
	src: '',
};

export const WithStatusIndicator: StoryFn<ComponentProps<typeof Avatar>> = (args) => (
	<div style={{ display: 'flex', alignItems: 'center', justifyContent: 'space-around' }}>
		<Avatar {...args} style={{ margin: '0.5rem' }} status='offline' />
		<Avatar {...args} style={{ margin: '0.5rem' }} status='away' />
		<Avatar {...args} style={{ margin: '0.5rem' }} status='busy' />
		<Avatar {...args} style={{ margin: '0.5rem' }} status='online' />
	</div>
);
WithStatusIndicator.storyName = 'with status indicator';
