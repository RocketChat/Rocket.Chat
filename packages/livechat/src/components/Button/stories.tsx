import { action } from '@storybook/addon-actions';
import type { Meta, StoryFn } from '@storybook/preact';
import type { ComponentProps } from 'preact';

import { Button } from '.';
import { gazzoAvatar } from '../../../.storybook/helpers';
import ChatIcon from '../../icons/chat.svg';

export default {
	title: 'components/Button',
	component: Button,
	args: {
		disabled: false,
		outline: false,
		nude: false,
		danger: false,
		secondary: false,
		stack: false,
		small: false,
		loading: false,
		badge: undefined,
		icon: null,
		img: undefined,
		children: 'Powered by Rocket.Chat',
		onClick: action('clicked'),
	},
	argTypes: {
		icon: {
			options: ['none', 'chat'],
			mapping: {
				none: null,
				chat: <ChatIcon />,
			},
		},
	},
	parameters: {
		layout: 'centered',
	},
} satisfies Meta<ComponentProps<typeof Button>>;

const Template: StoryFn<ComponentProps<typeof Button>> = (args) => <Button {...args} />;

export const Normal = Template.bind({});
Normal.storyName = 'normal';

export const Disabled = Template.bind({});
Disabled.storyName = 'disabled';
Disabled.args = {
	disabled: true,
};

export const Outline = Template.bind({});
Outline.storyName = 'outline';
Outline.args = {
	outline: true,
};

export const Nude = Template.bind({});
Nude.storyName = 'nude';
Nude.args = {
	nude: true,
};

export const Danger = Template.bind({});
Danger.storyName = 'danger';
Danger.args = {
	danger: true,
};

export const Secondary = Template.bind({});
Secondary.storyName = 'secondary';
Secondary.args = {
	secondary: true,
};

export const Stack = Template.bind({});
Stack.storyName = 'stack';
Stack.args = {
	stack: true,
};

export const Small = Template.bind({});
Small.storyName = 'small';
Small.args = {
	small: true,
};

export const Loading = Template.bind({});
Loading.storyName = 'loading';
Loading.args = {
	loading: true,
};

export const WithBadge = Template.bind({});
WithBadge.storyName = 'with badge';
WithBadge.args = {
	badge: 123,
};

export const WithIcon = Template.bind({});
WithIcon.storyName = 'with icon';
WithIcon.args = {
	icon: <ChatIcon />,
};

export const TransparentWithBackgroundImage = Template.bind({});
TransparentWithBackgroundImage.storyName = 'transparent with background image';
TransparentWithBackgroundImage.args = {
	img: gazzoAvatar,
	icon: <ChatIcon />,
};
