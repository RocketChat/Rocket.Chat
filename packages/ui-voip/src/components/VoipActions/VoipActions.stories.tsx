import type { Meta, StoryFn } from '@storybook/react';
import type { ReactElement } from 'react';

import VoipActions from './VoipActions';

const noop = () => undefined;

export default {
	title: 'Components/VoipActions',
	component: VoipActions,
	decorators: [(Story): ReactElement => <Story />],
} satisfies Meta<typeof VoipActions>;

export const IncomingActions: StoryFn<typeof VoipActions> = () => {
	return <VoipActions onDecline={noop} onAccept={noop} />;
};

export const OngoingActions: StoryFn<typeof VoipActions> = () => {
	return <VoipActions onEndCall={noop} onDTMF={noop} onHold={noop} onMute={noop} onTransfer={noop} />;
};

export const OutgoingActions: StoryFn<typeof VoipActions> = () => {
	return <VoipActions onEndCall={noop} />;
};
