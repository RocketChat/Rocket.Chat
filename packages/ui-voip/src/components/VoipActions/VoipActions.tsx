import { ButtonGroup } from '@rocket.chat/fuselage';
import { useTranslation } from 'react-i18next';

import ActionButton from '../VoipActionButton';

type VoipGenericActionsProps = {
	isDTMFActive?: boolean;
	isTransferActive?: boolean;
	isMuted?: boolean;
	isHeld?: boolean;
	onDTMF?: () => void;
	onTransfer?: () => void;
	onMute?: (muted: boolean) => void;
	onHold?: (held: boolean) => void;
};

type VoipIncomingActionsProps = VoipGenericActionsProps & {
	onEndCall?: never;
	onDecline: () => void;
	onAccept: () => void;
};

type VoipOngoingActionsProps = VoipGenericActionsProps & {
	onDecline?: never;
	onAccept?: never;
	onEndCall: () => void;
};

type VoipActionsProps = VoipIncomingActionsProps | VoipOngoingActionsProps;

const isIncoming = (props: VoipActionsProps): props is VoipIncomingActionsProps =>
	'onDecline' in props && 'onAccept' in props && !('onEndCall' in props);

const isOngoing = (props: VoipActionsProps): props is VoipOngoingActionsProps =>
	'onEndCall' in props && !('onAccept' in props && 'onDecline' in props);

const VoipActions = ({ isMuted, isHeld, isDTMFActive, isTransferActive, ...events }: VoipActionsProps) => {
	const { t } = useTranslation();

	return (
		<ButtonGroup large>
			{isIncoming(events) && <ActionButton danger label={t('Decline')} icon='phone-off' onClick={events.onDecline} />}

			<ActionButton
				label={isMuted ? t('Turn_on_microphone') : t('Turn_off_microphone')}
				icon='mic-off'
				pressed={isMuted}
				disabled={!events.onMute}
				onClick={() => events.onMute?.(!isMuted)}
			/>

			{!isIncoming(events) && (
				<ActionButton
					label={isHeld ? t('Resume') : t('Hold')}
					icon='pause-shape-unfilled'
					pressed={isHeld}
					disabled={!events.onHold}
					onClick={() => events.onHold?.(!isHeld)}
				/>
			)}

			<ActionButton
				label={isDTMFActive ? t('Close_Dialpad') : t('Open_Dialpad')}
				icon='dialpad'
				pressed={isDTMFActive}
				disabled={!events.onDTMF}
				onClick={events.onDTMF}
			/>

			<ActionButton
				label={t('Transfer_call')}
				icon='arrow-forward'
				pressed={isTransferActive}
				disabled={!events.onTransfer}
				onClick={events.onTransfer}
			/>

			{isOngoing(events) && <ActionButton danger label={t('End_call')} icon='phone-off' disabled={isHeld} onClick={events.onEndCall} />}

			{isIncoming(events) && <ActionButton success label={t('Accept')} icon='phone' onClick={events.onAccept} />}
		</ButtonGroup>
	);
};

export default VoipActions;
