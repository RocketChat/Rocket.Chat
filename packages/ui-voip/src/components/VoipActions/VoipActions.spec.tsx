import { mockAppRoot } from '@rocket.chat/mock-providers';
import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import * as stories from './VoipActions.stories';

const testCases = Object.values(composeStories(stories)).map((story) => [story.storyName || 'Story', story]);

test.each(testCases)(`renders %s without crashing`, async (_storyname, Story) => {
	const tree = render(<Story />, { wrapper: mockAppRoot().build() });
	expect(tree.baseElement).toMatchSnapshot();
});

test.each(testCases)('%s should have no a11y violations', async (_storyname, Story) => {
	const { container } = render(<Story />, { wrapper: mockAppRoot().build() });

	const results = await axe(container);
	expect(results).toHaveNoViolations();
});
