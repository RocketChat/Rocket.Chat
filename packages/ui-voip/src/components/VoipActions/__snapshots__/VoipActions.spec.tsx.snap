// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders IncomingActions without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-button-group rcx-button-group--align-start rcx-button-group--large"
      role="group"
    >
      <button
        aria-label="Decline"
        class="rcx-box rcx-box--full rcx-button--medium-square rcx-button--icon-secondary-danger rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
        title="Decline"
        type="button"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-phone-off rcx-icon"
        >
          
        </i>
      </button>
      <button
        aria-label="Turn_off_microphone"
        class="rcx-box rcx-box--full rcx-button--medium-square rcx-button--icon-secondary rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
        disabled=""
        title="Turn_off_microphone"
        type="button"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-mic-off rcx-icon"
        >
          
        </i>
      </button>
      <button
        aria-label="Open_Dialpad"
        class="rcx-box rcx-box--full rcx-button--medium-square rcx-button--icon-secondary rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
        disabled=""
        title="Open_Dialpad"
        type="button"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-dialpad rcx-icon"
        >
          
        </i>
      </button>
      <button
        aria-label="Transfer_call"
        class="rcx-box rcx-box--full rcx-button--medium-square rcx-button--icon-secondary rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
        disabled=""
        title="Transfer_call"
        type="button"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-arrow-forward rcx-icon"
        >
          
        </i>
      </button>
      <button
        aria-label="Accept"
        class="rcx-box rcx-box--full rcx-button--medium-square rcx-button--icon-secondary-success rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
        title="Accept"
        type="button"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-phone rcx-icon"
        >
          
        </i>
      </button>
    </div>
  </div>
</body>
`;

exports[`renders OngoingActions without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-button-group rcx-button-group--align-start rcx-button-group--large"
      role="group"
    >
      <button
        aria-label="Turn_off_microphone"
        class="rcx-box rcx-box--full rcx-button--medium-square rcx-button--icon-secondary rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
        title="Turn_off_microphone"
        type="button"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-mic-off rcx-icon"
        >
          
        </i>
      </button>
      <button
        aria-label="Hold"
        class="rcx-box rcx-box--full rcx-button--medium-square rcx-button--icon-secondary rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
        title="Hold"
        type="button"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-pause-shape-unfilled rcx-icon"
        >
          
        </i>
      </button>
      <button
        aria-label="Open_Dialpad"
        class="rcx-box rcx-box--full rcx-button--medium-square rcx-button--icon-secondary rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
        title="Open_Dialpad"
        type="button"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-dialpad rcx-icon"
        >
          
        </i>
      </button>
      <button
        aria-label="Transfer_call"
        class="rcx-box rcx-box--full rcx-button--medium-square rcx-button--icon-secondary rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
        title="Transfer_call"
        type="button"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-arrow-forward rcx-icon"
        >
          
        </i>
      </button>
      <button
        aria-label="End_call"
        class="rcx-box rcx-box--full rcx-button--medium-square rcx-button--icon-secondary-danger rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
        title="End_call"
        type="button"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-phone-off rcx-icon"
        >
          
        </i>
      </button>
    </div>
  </div>
</body>
`;

exports[`renders OutgoingActions without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-button-group rcx-button-group--align-start rcx-button-group--large"
      role="group"
    >
      <button
        aria-label="Turn_off_microphone"
        class="rcx-box rcx-box--full rcx-button--medium-square rcx-button--icon-secondary rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
        disabled=""
        title="Turn_off_microphone"
        type="button"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-mic-off rcx-icon"
        >
          
        </i>
      </button>
      <button
        aria-label="Hold"
        class="rcx-box rcx-box--full rcx-button--medium-square rcx-button--icon-secondary rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
        disabled=""
        title="Hold"
        type="button"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-pause-shape-unfilled rcx-icon"
        >
          
        </i>
      </button>
      <button
        aria-label="Open_Dialpad"
        class="rcx-box rcx-box--full rcx-button--medium-square rcx-button--icon-secondary rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
        disabled=""
        title="Open_Dialpad"
        type="button"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-dialpad rcx-icon"
        >
          
        </i>
      </button>
      <button
        aria-label="Transfer_call"
        class="rcx-box rcx-box--full rcx-button--medium-square rcx-button--icon-secondary rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
        disabled=""
        title="Transfer_call"
        type="button"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-arrow-forward rcx-icon"
        >
          
        </i>
      </button>
      <button
        aria-label="End_call"
        class="rcx-box rcx-box--full rcx-button--medium-square rcx-button--icon-secondary-danger rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
        title="End_call"
        type="button"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-phone-off rcx-icon"
        >
          
        </i>
      </button>
    </div>
  </div>
</body>
`;
